mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -156,7 +160,22 @@ private ParsedAttachment retrieveAttachment(Entity entity) throws IOException {
     }
 
     private Optional<String> contentType(Optional<ContentTypeField> contentTypeField) {
-        return contentTypeField.map(ContentTypeField::getMimeType);
+        return contentTypeField.map(this::contentTypePreserveCharset);
```

Comment:
why only charset must be preserved? Have you a rfc reference somewhere?

Reply:
https://jmap.io/spec-core.html#uploading-binary-data

`type: String The media type of the file (as specified in [@!RFC6838], Section 4.2) as set in the Content-Type header of the upload HTTP request.`

Looks like it should be the over way around: only keep the mimeType and always discard other informations.